cloudformation/membership-attribute-service.yaml

```diff
@@ -0,0 +1,329 @@
+            Effect: Allow
+          - Action:
+            - cloudwatch:*
+            - logs:*
+            Resource: "*"
+            Effect: Allow
+  InstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Path: "/"
+      Roles:
+      - Ref: MembershipRole
+  GuardianAccessSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: SSH access from Guardian network
+      VpcId:
+        Ref: VpcId
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: '22'
+        ToPort: '22'
+        CidrIp: 77.91.248.0/21
+  LoadBalancer:
+    Type: AWS::ElasticLoadBalancing::LoadBalancer
+    Properties:
+      Listeners:
+      - LoadBalancerPort: '443'
+        InstancePort: '9000'
+        Protocol: HTTPS
+        SSLCertificateId: !Sub
+          - "arn:aws:iam::${AWS::AccountId}:server-certificate/${SiteDomain}"
+          - {}
+      ConnectionDrainingPolicy:
+        Enabled: 'true'
+        Timeout: '60'
+      CrossZone: 'true'
+      HealthCheck:
+        Target: HTTP:9000/healthcheck
+        HealthyThreshold: '2'
+        UnhealthyThreshold: '3'
+        Interval: '10'
+        Timeout: '5'
+      Subnets:
+        Ref: VpcSubnets
+      SecurityGroups:
+      - Ref: LoadBalancerSecurityGroup
+  AutoscalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      AvailabilityZones:
+        Fn::GetAZs: ''
+      LaunchConfigurationName:
+        Ref: LaunchConfig
+      MinSize:
+        Ref: MinInstances
+      MaxSize:
+        Ref: MaxInstances
+      HealthCheckType: ELB
+      HealthCheckGracePeriod: 400
+      LoadBalancerNames:
+      - Ref: LoadBalancer
+      Tags:
+      - Key: Stage
+        Value:
+          Ref: Stage
+        PropagateAtLaunch: 'true'
+      - Key: PlayAppSecret
+        Value:
+          Ref: PlayAppSecret
+        PropagateAtLaunch: 'true'
+      - Key: Name
+        Value:
+          Fn::FindInMap: [ StageVariables, { Ref: Stage }, InstanceName ]
+        PropagateAtLaunch: 'true'
+      - Key: Stack
+        Value: membership
+        PropagateAtLaunch: 'true'
+      - Key: App
+        Value: membership-attribute-service
+        PropagateAtLaunch: 'true'
+      - Key: Role
+        Value: membership-attribute-service
+        PropagateAtLaunch: 'true'
+      - Key: Mainclass
+        Value: membership-attribute-service
+        PropagateAtLaunch: 'true'
+      VPCZoneIdentifier:
+        Ref: VpcSubnets
+  LaunchConfig:
+    Type: AWS::AutoScaling::LaunchConfiguration
+    Properties:
+      KeyName:
+        Ref: KeyName
+      ImageId:
+        Ref: AmiId
+      SecurityGroups:
+      - Ref: InstanceSecurityGroup
+      - Ref: GuardianAccessSecurityGroup
+      - Ref: VulnerabilityScanningSecurityGroup
+      InstanceType:
+        Ref: InstanceType
+      AssociatePublicIpAddress: 'True'
+      IamInstanceProfile:
+        Ref: InstanceProfile
+      UserData:
+        "Fn::Base64":
+          !Sub
```

Comment:
I used json2yaml.com for most of it, and then manually tidied up the UserData to use the !Sub, which is so much easier to read.